nixos/modules/services/misc/gerrit.nix

```diff
@@ -0,0 +1,158 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.gerrit;
+
+  cmd = "${pkgs.jre_headless}/bin/java -jar ${cfg.package}/webapps/gerrit-${cfg.package.version}.war";
+
+  mkPluginSectionName = plugin: ''plugin "${escape [ "\"" ] plugin}"'';
+
+  settingsFile = pkgs.writeText "gerrit.config" ''
+    ${generators.toINI {} cfg.settings}
+    ${generators.toINI { mkSectionName = mkPluginSectionName; } cfg.pluginSettings}
+  '';
+
+  pluginDir = pkgs.runCommandNoCC "gerrit-plugins"
+    { inherit (cfg) plugins; }
+    ''
+      mkdir $out
+      cd $out
+      for plugin in $plugins; do
+        ln -sv $plugin
+      done
+    '';
+
+in
+
+{
+
+  options = {
+
+    services.gerrit = {
+
+      enable = mkEnableOption "Gerrit Code Review";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.gerrit;
+        defaultText = "pkgs.gerrit";
+        description = ''
+          The package used for the Gerrit daemon.
+        '';
+      };
+
+      settings = mkOption {
+        type = with types; lazyAttrsOf (lazyAttrsOf str);
+        default = {};
+      };
+
+      basePath = mkOption {
+        type = types.str;
+        default = "git";
+        description = ''
+          Local filesystem directory holding all Git repositories that Gerrit 
+          knows about and can process changes for. A project entity in Gerrit 
+          maps to a local Git repository by creating the path string 
+          "''${basePath}/''${project_name}.git".
+
+          If relative, the path is resolved relative to /var/lib/gerrit.
+        '';
+      };
+
+      canonicalWebUrl = mkOption {
+        type = types.str;
+        description = ''
+          The default URL for Gerrit to be accessed through.
+          Typically this would be set to something like 
+          "http://review.example.com/" or "http://example.com:8080/gerrit/" so 
+          Gerrit can output links that point back to itself.
+        '';
+      };
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "0.0.0.0:80";
```

Comment:
I'm not entirely sure if exposing http to the internet on port 80 is ever a good idea. Is the intended usage behind some https reverse proxy? Or on an internal net?

Reply:
It depends on your setup. The machine might be forwarded from an internal or external load-balancer. This is up to the user really.

The default when installing gerrit is  `*:8080`

Reply:
I've been using this behind an HTTPS reverse proxy. I suppose this should plausibly default to interacting with the ACME module, but I personally prefer to keep the actual TLS keys away from services themselves. My production deployment uses Google Cloud's TLS termination.